.github/workflows/cmake.yml

```diff
@@ -0,0 +1,62 @@
+name: CMake
+
+on:
+  push:
+    branches: [ "main" ]
```

Comment:
Minor: I'd prefer:

```
branches: 
  - main
```

With each branch in its own line, it's easier to diff when you add new items.